server/server.go

```diff
@@ -266,9 +264,13 @@ func (s *IDPServer) newMux() *http.ServeMux {
 	// Register /register endpoint for Dynamic Client Registration
 	mux.HandleFunc("/register", s.addGrantAccessContext(s.serveDynamicClientRegistration))
 
+	// Register /clients/ - API access to manage clients DB
+	// wrap it in a cross origin protection handler to prevent CSRF
+	mux.HandleFunc("/clients/", s.csrfCheck(s.addGrantAccessContext(s.serveClients)))
```

Comment:
you could probably add the CSRF wrapper to all endpoints (around the `mux`).
all GET/HEAD/OPTIONS requests will be exempt.
all POST requests will only trigger the check for browser requests (that include the `Origin` header), so should be a noop for programmatic requests.